core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -345,7 +347,29 @@ class BrokerServer(
       tokenManager = new DelegationTokenManager(config, tokenCache, time)
       tokenManager.startup()
 
-      shareCoordinator = createShareCoordinator()
+      /* get persister instance */
+      persister = if (config.shareGroupConfig.shareGroupPersisterClassName.nonEmpty) {
+        val klass = Utils.loadClass(config.shareGroupConfig.shareGroupPersisterClassName, classOf[Object]).asInstanceOf[Class[Persister]]
+        klass.getDeclaredMethod("getInstance")
```

Comment:
@junrao Some issues in going with Authorizer approach.

- The persister is a singleton object (private constructor). As it stands currently, persister is instantiated in `BrokerServer` and passed to `SharePartitionManager`. The same instance will also be passed to group-coordinator in future.
- Secondly, there are a few items which need to be passed to persister which can only be instantiated in the core module (namely network client and cache helper). So it not possible to have a separate Util class ins `share` module which can be delegated to by `BrokerServer` to do the work.
- Regarding the configuration - as mentioned above, we need to pass some objects which can only be instantiated in `core`, if we use the `Configurable` interface we will have to pass them via a map of <String, ?>. This will require casting while retrieving. I was initially planning to use this but then created a custom configure method in Persister interface as it seemed neater and we can add and retrieve additional configs without casting.
- I can go with `Configurable` insterface if you feel that is more appropriate?

Reply:
The comment of Persister interface says the following.

> but this interface allows for other variations as well.

However, the current interface makes it impossible for a different implementation. So do we still want the Persister to be pluggable?

Reply:
After some discussion we have concluded that the persister will be Factory pluggable. That is will an internal config which will not be exposed to the user.
The implementations will be internal.